File: cdk-from-cfn-testing/expected/batch/golang/BatchStack.go
package main

import (
	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	batch "github.com/aws/aws-cdk-go/awscdk/v2/awsbatch"
	ec2 "github.com/aws/aws-cdk-go/awscdk/v2/awsec2"
	iam "github.com/aws/aws-cdk-go/awscdk/v2/awsiam"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type BatchStackProps struct {
	cdk.StackProps
}

/// AWS CloudFormation Sample Template Managed Single Batch Job Queue: This template demonstrates the usage of simple Job Queue and EC2 style Compute Environment.  **WARNING** You will be billed for the AWS resources used if you create a stack from this template.
type BatchStack struct {
	cdk.Stack
	ComputeEnvironmentArn interface{} // TODO: fix to appropriate type
	JobQueueArn interface{} // TODO: fix to appropriate type
	JobDefinitionArn interface{} // TODO: fix to appropriate type
}

func NewBatchStack(scope constructs.Construct, id string, props *BatchStackProps) *BatchStack {
	var sprops cdk.StackProps
	if props != nil {
		sprops = props.StackProps
	}
	stack := cdk.NewStack(scope, &id, &sprops)

	batchServiceRole := iam.NewCfnRole(
		stack,
		jsii.String("BatchServiceRole"),
		&iam.CfnRoleProps{
			AssumeRolePolicyDocument: map[string]interface{} {
				"Version": jsii.String("2012-10-17"),
				"Statement": &[]interface{}{
					map[string]interface{} {
						"Effect": jsii.String("Allow"),
						"Principal": map[string]interface{} {
							"Service": jsii.String("batch.amazonaws.com"),
						},
						"Action": jsii.String("sts:AssumeRole"),
					},
				},
			},
			ManagedPolicyArns: &[]*string{
				jsii.String("arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole"),
			},
		},
	)

	ecsInstanceRole := iam.NewCfnRole(
		stack,
		jsii.String("EcsInstanceRole"),
		&iam.CfnRoleProps{
			AssumeRolePolicyDocument: map[string]interface{} {
				"Version": jsii.String("2008-10-17"),
				"Statement": &[]interface{}{
					map[string]interface{} {
						"Sid": jsii.String(""),
						"Effect": jsii.String("Allow"),
						"Principal": map[string]interface{} {
							"Service": jsii.String("ec2.amazonaws.com"),
						},
						"Action": jsii.String("sts:AssumeRole"),
					},
				},
			},
			ManagedPolicyArns: &[]*string{
				jsii.String("arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"),
			},
		},
	)

	internetGateway := ec2.NewCfnInternetGateway(
		stack,
		jsii.String("InternetGateway"),
		&ec2.CfnInternetGatewayProps{
		},
	)

	jobDefinition := batch.NewCfnJobDefinition(
		stack,
		jsii.String("JobDefinition"),
		&batch.CfnJobDefinitionProps{
			Type: jsii.String("container"),
			ContainerProperties: &ContainerPropertiesProperty{
				Image: cdk.Fn_Join(jsii.String(""), &[]*string{
					jsii.String("137112412989.dkr.ecr."),
					stack.Region(),
					jsii.String(".amazonaws.com/amazonlinux:latest"),
				}),
				Vcpus: jsii.Number(2),
				Memory: jsii.Number(2000),
				Command: &[]*string{
					jsii.String("echo"),
					jsii.String("Hello world"),
				},
			},
			RetryStrategy: &RetryStrategyProperty{
				Attempts: jsii.Number(1),
			},
		},
	)

	vpc := ec2.NewCfnVPC(
		stack,
		jsii.String("VPC"),
		&ec2.CfnVPCProps{
			CidrBlock: jsii.String("10.0.0.0/16"),
		},
	)

	iamInstanceProfile := iam.NewCfnInstanceProfile(
		stack,
		jsii.String("IamInstanceProfile"),
		&iam.CfnInstanceProfileProps{
			Roles: &[]*string{
				ecsInstanceRole.Ref(),
			},
		},
	)

	routeTable := ec2.NewCfnRouteTable(
		stack,
		jsii.String("RouteTable"),
		&ec2.CfnRouteTableProps{
			VpcId: vpc.Ref(),
		},
	)

	securityGroup := ec2.NewCfnSecurityGroup(
		stack,
		jsii.String("SecurityGroup"),
		&ec2.CfnSecurityGroupProps{
			GroupDescription: jsii.String("EC2 Security Group for instances launched in the VPC by Batch"),
			VpcId: vpc.Ref(),
		},
	)

	subnet := ec2.NewCfnSubnet(
		stack,
		jsii.String("Subnet"),
		&ec2.CfnSubnetProps{
			CidrBlock: jsii.String("10.0.0.0/24"),
			VpcId: vpc.Ref(),
			MapPublicIpOnLaunch: jsii.Bool(true),
		},
	)

	ec2.NewCfnVPCGatewayAttachment(
		stack,
		jsii.String("VPCGatewayAttachment"),
		&ec2.CfnVPCGatewayAttachmentProps{
			VpcId: vpc.Ref(),
			InternetGatewayId: internetGateway.Ref(),
		},
	)

	computeEnvironment := batch.NewCfnComputeEnvironment(
		stack,
		jsii.String("ComputeEnvironment"),
		&batch.CfnComputeEnvironmentProps{
			Type: jsii.String("MANAGED"),
			ComputeResources: &ComputeResourcesProperty{
				Type: jsii.String("EC2"),
				MinvCpus: jsii.Number(0),
				DesiredvCpus: jsii.Number(0),
				MaxvCpus: jsii.Number(64),
				InstanceTypes: &[]*string{
					jsii.String("optimal"),
				},
				Subnets: &[]*string{
					subnet.Ref(),
				},
				SecurityGroupIds: &[]*string{
					securityGroup.Ref(),
				},
				InstanceRole: iamInstanceProfile.Ref(),
			},
			ServiceRole: batchServiceRole.Ref(),
		},
	)

	ec2.NewCfnRoute(
		stack,
		jsii.String("Route"),
		&ec2.CfnRouteProps{
			RouteTableId: routeTable.Ref(),
			DestinationCidrBlock: jsii.String("0.0.0.0/0"),
			GatewayId: internetGateway.Ref(),
		},
	)

	ec2.NewCfnSubnetRouteTableAssociation(
		stack,
		jsii.String("SubnetRouteTableAssociation"),
		&ec2.CfnSubnetRouteTableAssociationProps{
			RouteTableId: routeTable.Ref(),
			SubnetId: subnet.Ref(),
		},
	)

	jobQueue := batch.NewCfnJobQueue(
		stack,
		jsii.String("JobQueue"),
		&batch.CfnJobQueueProps{
			Priority: jsii.Number(1),
			ComputeEnvironmentOrder: &[]interface{}{
				&ComputeEnvironmentOrderProperty{
					Order: jsii.Number(1),
					ComputeEnvironment: computeEnvironment.Ref(),
				},
			},
		},
	)

	return &BatchStack{
		Stack: stack,
		ComputeEnvironmentArn: computeEnvironment.Ref(),
		JobQueueArn: jobQueue.Ref(),
		JobDefinitionArn: jobDefinition.Ref(),
	}
}

File: cdk-from-cfn-testing/expected/batch/typescript/BatchStack.ts
import * as cdk from 'aws-cdk-lib';
import * as batch from 'aws-cdk-lib/aws-batch';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import * as iam from 'aws-cdk-lib/aws-iam';

export interface BatchStackProps extends cdk.StackProps {
}

/**
 * AWS CloudFormation Sample Template Managed Single Batch Job Queue: This template demonstrates the usage of simple Job Queue and EC2 style Compute Environment.  **WARNING** You will be billed for the AWS resources used if you create a stack from this template.
 */
export class BatchStack extends cdk.Stack {
  public readonly computeEnvironmentArn;
  public readonly jobQueueArn;
  public readonly jobDefinitionArn;

  public constructor(scope: cdk.App, id: string, props: BatchStackProps = {}) {
    super(scope, id, props);

    // Resources
    const batchServiceRole = new iam.CfnRole(this, 'BatchServiceRole', {
      assumeRolePolicyDocument: {
        Version: '2012-10-17',
        Statement: [
          {
            Effect: 'Allow',
            Principal: {
              Service: 'batch.amazonaws.com',
            },
            Action: 'sts:AssumeRole',
          },
        ],
      },
      managedPolicyArns: [
        'arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole',
      ],
    });

    const ecsInstanceRole = new iam.CfnRole(this, 'EcsInstanceRole', {
      assumeRolePolicyDocument: {
        Version: '2008-10-17',
        Statement: [
          {
            Sid: '',
            Effect: 'Allow',
            Principal: {
              Service: 'ec2.amazonaws.com',
            },
            Action: 'sts:AssumeRole',
          },
        ],
      },
      managedPolicyArns: [
        'arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role',
      ],
    });

    const internetGateway = new ec2.CfnInternetGateway(this, 'InternetGateway', {
    });

    const jobDefinition = new batch.CfnJobDefinition(this, 'JobDefinition', {
      type: 'container',
      containerProperties: {
        image: [
          '137112412989.dkr.ecr.',
          this.region,
          '.amazonaws.com/amazonlinux:latest',
        ].join(''),
        vcpus: 2,
        memory: 2000,
        command: [
          'echo',
          'Hello world',
        ],
      },
      retryStrategy: {
        attempts: 1,
      },
    });

    const vpc = new ec2.CfnVPC(this, 'VPC', {
      cidrBlock: '10.0.0.0/16',
    });

    const iamInstanceProfile = new iam.CfnInstanceProfile(this, 'IamInstanceProfile', {
      roles: [
        ecsInstanceRole.ref,
      ],
    });

    const routeTable = new ec2.CfnRouteTable(this, 'RouteTable', {
      vpcId: vpc.ref,
    });

    const securityGroup = new ec2.CfnSecurityGroup(this, 'SecurityGroup', {
      groupDescription: 'EC2 Security Group for instances launched in the VPC by Batch',
      vpcId: vpc.ref,
    });

    const subnet = new ec2.CfnSubnet(this, 'Subnet', {
      cidrBlock: '10.0.0.0/24',
      vpcId: vpc.ref,
      mapPublicIpOnLaunch: true,
    });

    const vpcGatewayAttachment = new ec2.CfnVPCGatewayAttachment(this, 'VPCGatewayAttachment', {
      vpcId: vpc.ref,
      internetGatewayId: internetGateway.ref,
    });

    const computeEnvironment = new batch.CfnComputeEnvironment(this, 'ComputeEnvironment', {
      type: 'MANAGED',
      computeResources: {
        type: 'EC2',
        minvCpus: 0,
        desiredvCpus: 0,
        maxvCpus: 64,
        instanceTypes: [
          'optimal',
        ],
        subnets: [
          subnet.ref,
        ],
        securityGroupIds: [
          securityGroup.ref,
        ],
        instanceRole: iamInstanceProfile.ref,
      },
      serviceRole: batchServiceRole.ref,
    });

    const route = new ec2.CfnRoute(this, 'Route', {
      routeTableId: routeTable.ref,
      destinationCidrBlock: '0.0.0.0/0',
      gatewayId: internetGateway.ref,
    });

    const subnetRouteTableAssociation = new ec2.CfnSubnetRouteTableAssociation(this, 'SubnetRouteTableAssociation', {
      routeTableId: routeTable.ref,
      subnetId: subnet.ref,
    });

    const jobQueue = new batch.CfnJobQueue(this, 'JobQueue', {
      priority: 1,
      computeEnvironmentOrder: [
        {
          order: 1,
          computeEnvironment: computeEnvironment.ref,
        },
      ],
    });

    // Outputs
    this.computeEnvironmentArn = computeEnvironment.ref;
    new cdk.CfnOutput(this, 'CfnOutputComputeEnvironmentArn', {
      key: 'ComputeEnvironmentArn',
      value: this.computeEnvironmentArn!.toString(),
    });
    this.jobQueueArn = jobQueue.ref;
    new cdk.CfnOutput(this, 'CfnOutputJobQueueArn', {
      key: 'JobQueueArn',
      value: this.jobQueueArn!.toString(),
    });
    this.jobDefinitionArn = jobDefinition.ref;
    new cdk.CfnOutput(this, 'CfnOutputJobDefinitionArn', {
      key: 'JobDefinitionArn',
      value: this.jobDefinitionArn!.toString(),
    });
  }
}
